ghcide/src/Development/IDE/Plugin/Completions.hs

```diff
@@ -77,10 +79,18 @@ produceCompletions = do
               let env = hscEnv sess
               -- We do this to be able to provide completions of items that are not restricted to the explicit list
               res <- liftIO $ tcRnImportDecls env (dropListFromImportDecl <$> imps)
-              case res of
-                  (_, Just rdrEnv) -> do
+              inScope <- liftIO $ tcRnImportDecls env imps
+              case (res, inScope) of
+                  ((_, Just globalEnv), (_, Just inScopeEnv)) -> do
                       let uri = fromNormalizedUri $ normalizedFilePathToUri file
-                      cdata <- liftIO $ cacheDataProducer uri env (ms_mod ms) rdrEnv imps parsedDeps
+                          globalElts = globalRdrEnvElts globalEnv
+                          inScopeElts = globalRdrEnvElts inScopeEnv
+                          diff = getDiffBy (\GRE {gre_name = nameA} GRE {gre_name = nameB} -> nameA == nameB) globalElts inScopeElts
+                          rdrElts = flip map diff $ \case
+                            First x -> (x, False)
+                            Second x -> (x, True)
+                            Both x _ -> (x, True)
```

Comment:
Let's roll this back, benchmarks show that this diff computation is rather expensive.
Sorry, I didn't realise that `OccEnv a` doesn't provide any means of taking the set of keys.

This is a local benchmark, where:

HEAD: f8c796f112a9680db6ab85a68d7bd592c5a9dc7e
no-diff: e2067a365c863a00d31f2f66432d25a2219c007b

```
diff --git a/ghcide/bench/config.yaml b/ghcide/bench/config.yaml
index 25cf4080..43b1675b 100644
--- a/ghcide/bench/config.yaml
+++ b/ghcide/bench/config.yaml
@@ -62,5 +62,6 @@ versions:
 # - ghcide-v0.7.1
 # - ghcide-v0.7.2
 # - ghcide-v0.7.3
+- no-diff: e2067a365c863a00d31f2f66432d25a2219c007b
 - HEAD
(END)
```

<img width="798" alt="image" src="https://user-images.githubusercontent.com/26626/107081780-21625500-67eb-11eb-929d-ab4569cabb90.png">